### PR TITLE
sof-logger: add checksums to -d(ump_file) option

### PR DESCRIPTION
**Pull request by marc-hb**
3 commits

Sample output:
```
logger ABI Version is   5:3:0
ldc_file ABI Version is 5:3:0
ldc_file src checksum           0x07d4f1ad       <=== NEW!
Loaded FW expects chksum        0x07d4f1ad       <=== NEW!

Components uuid dictionary size:        2400 bytes
Components uuid base address:           0x1fffa000
...
```

### CODE REVIEW

**Comment by marc-hb** [comment]
Everything is green except this checkpatch warning which was already there before this PR:
```
WARNING: quoted string split across lines
#74: FILE: tools/logger/logger.c:47:
+	fprintf(stdout, "%s:\t -v ver_file\t\tUse ver_file instead of "
+		"/sys/kernel/debug/sof/fw_version\n", APP_NAME);
```
I suspect the purpose of this warning is to make sure error messages can be `git grep`ed. Not really an issue here.

File: tools/logger/logger.c [hunk 1]
```diff
@@ -270,7 +270,6 @@ int main(int argc, char *argv[])
 				usage();
 			}
 			config.dump_ldc = 1;
-			config.version_fw = 0;
```
Comment (ktrzcinx) [inline tools/logger/logger.c:273]: What motivation stands behind this change ?

Comment (marc-hb) [reply tools/logger/logger.c:273]: Thanks @ktrzcinx for reviewing this PR, really appreciated.

This is what allows the `-d ldc_file` option to ALSO show the checksum expected by the firmware - when and only when a firmware is running. See sample output in the description. I suggest looking at each commit independently if you have looked only at the entire diff for now.

Comment (ktrzcinx) [reply tools/logger/logger.c:273]: Ok, I've got it!

**Review by kv2019i — APPROVED** [review]

**Comment by ujfalusi** [comment]
Does the duplicated checksum gives anything for the user?
```
ldc_file src checksum           0x07d4f1ad       <=== NEW!
Loaded FW expects chksum        0x07d4f1ad       <=== NEW!
```
Would not be better to print a single line of the two checksum matches:
```
ldc checksum: 0x07d4f1ad
```
If there is a mismatch between the two checksum then:
```
lcd_file src checksum is 0x07d4f1ad while the loaded firmware expects 0x1b67cba9
```

**Comment by marc-hb** [comment]
> Would not be better to print a single line of the two checksum matches:

I don't think so because a single line is also printed when the firmware is not loaded.

> If there is a mismatch between the two checksum then:
> lcd_file src checksum is 0x07d4f1ad while the loaded firmware expects 0x1b67cba9

I personally prefer having the checksums close to each other:

```
ldc_file src checksum          0x07d4f1ad
Loaded FW expects chksum       0x1b67cba9
```
... and this makes the code simpler because it does not have to compare them.

File: tools/logger/convert.c [hunk 2]
```diff
@@ -950,13 +950,16 @@ static int logger_read(void)
...
+ * /sys/kernel/debug/sof/fw_version) with the checksum in the .ldc file.
+ * @return 0 when the checksums match
+ */
+static int verify_ldc_checksum(const uint32_t ldc_sum)
 {
 	struct sof_ipc_fw_version ver;
 	int count;
 
-	if (!global_config->version_fd)
+	if (!global_config->version_fw) // -n option
```
Comment (lgirdwood) [inline tools/logger/convert.c]: C style

Comment (marc-hb) [reply tools/logger/convert.c]: Wait, I thought #5046 meant this is allowed now?

Comment (lgirdwood) [reply tools/logger/convert.c]: Fine in the SPDX, but my pattern matching needs C style in the code.

Comment (marc-hb) [reply tools/logger/convert.c]: Done.

**Comment by marc-hb** [comment]
Unusual, isolated and totally unrelated timeout in https://sof-ci.01.org/sofpr/PR5110/build11443/devicetest/?model=ADLP_RVP_NOCODEC&testcase=check-pause-resume-capture-10, everything else is green.

**Review by lyakh — APPROVED** [review]

**Review by ujfalusi — APPROVED** [review]

**Review by ktrzcinx — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**